### PR TITLE
Feat/refacto header config

### PR DESCRIPTION
**Pull request by VirAlex**
## Objective
What: It was not enough personalisable for example if we wanted use filters with more config:
[Filters](https://www.algolia.com/doc/api-reference/api-parameters/filters/?client=javascript)
Why: SE's Request
How: With a new attribute in config

## Type

- [x] New Feature

## Documented

- [x] Have you documented in changed file using comments

### CODE REVIEW

File: src/config/navigationConfig.js [hunk 1]
```diff
@@ -37,6 +39,14 @@ export const linksHeader = atom({
       filter: 'Womens',
       url: '/womens',
     },
+    {
+      name: 'Womens and Accessories Even and Odd',
+      type: 'filterComplex',
+      filter: '',
+      moreComplexFilter:
```
Comment (matthew1809) [inline src/config/navigationConfig.js:46]: should we adjust the attribute name to be something clearer like `rawFilter` and make that the type as well?

File: src/config/navigationConfig.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ export const categoryPageFilterAttribute = 'hierarchicalCategories.lvl0';
 // The name is for display purposes for each navigation title
 // The filter is the value to filter on to show results for the linked category page
 // The url should be the lower case URL friendly version of the filter value
+// Please use filterComplex type and moreComplexFilter attribute to use the filter with complexity
```
Comment (matthew1809) [inline src/config/navigationConfig.js:18]: You can say instead something like "if you need to define your own filter attributes and values, use rawFilters, instead use "filter". Remember that filter will use the categoryPageAttribute."

**Review by matthew1809 — CHANGES_REQUESTED** [review]
Great job man! Some some renaming suggestions just for clarity and consistancy

File: src/config/navigationConfig.js (hunk 2 above)
Comment (VirAlex) [reply src/config/navigationConfig.js:18]: Thank you @matthew1809 for the correction 🙏

Comment (VirAlex) [reply src/config/navigationConfig.js:18]: Do you think I have to change key name in the object? 
Screenshot 2022-08-24 at 12 29 18

**Review by matthew1809 — APPROVED** [review]
GG!

**Pull request merged**